src/metal.cs

```diff
@@ -699,6 +793,94 @@ partial interface MTLDevice {
 #endif
 		[Export ("supportsTextureSampleCount:")]
 		bool SupportsTextureSampleCount (nuint sampleCount);
+
+		[Mac (10, 13), NoiOS, NoWatch, NoTV]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("removable")]
+		bool Removable { [Bind ("isRemovable")] get; }
+
+		[Mac (10, 13), NoiOS, NoWatch, NoTV]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("readWriteTextureSupport")]
+		MTLReadWriteTextureTier ReadWriteTextureSupport { get; }
+
+		// @required @property (readonly) MTLArgumentBuffersTier argumentBuffersSupport __attribute__((availability(macos, introduced=10_13)));
+		[Mac (10, 13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("argumentBuffersSupport")]
+		MTLArgumentBuffersTier ArgumentBuffersSupport { get; }
+
+		// @required @property (readonly, getter = areRasterOrderGroupsSupported) BOOL rasterOrderGroupsSupported __attribute__((availability(macos, introduced=10_13)));
+		[Mac (10, 13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("rasterOrderGroupsSupported")]
+		bool RasterOrderGroupsSupported { [Bind ("areRasterOrderGroupsSupported")] get; }
+
+		[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("newLibraryWithURL:error:")]
+		[return: NullAllowed]
+		IMTLLibrary CreateLibrary (NSUrl url, [NullAllowed] out NSError error);
+
+		[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("minimumLinearTextureAlignmentForPixelFormat:")]
+		nuint GetMinimumLinearTextureAlignment (MTLPixelFormat format);
+
+		[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("maxThreadgroupMemoryLength")]
+		nuint MaxThreadgroupMemoryLength { get; }
+
+[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("programmableSamplePositionsSupported")]
+		bool ProgrammableSamplePositionsSupported { [Bind ("areProgrammableSamplePositionsSupported")] get; }
+
+		[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("getDefaultSamplePositions:count:")]
+		void GetDefaultSamplePositions (IntPtr positions, nuint count);
```

Comment:
I did add extension methods for the IntPtr bindings you pointed out.  The one for GetDefaultSamplePositions is at https://github.com/xamarin/xamarin-macios/pull/2457/files#diff-3ca0bc76e19315f7d067e894abf82f1fR11

I added them in https://github.com/xamarin/xamarin-macios/pull/2457/commits/b9279a2dd7783d8b02a9f128ee407eecace97c4c, though they got some minor modifications (mostly removing duplicate unsafe) in the following commit